src/cpp/rtps/reader/CompoundReaderListener.cpp

```diff
@@ -44,6 +44,7 @@ bool CompoundReaderListener::hasReaderAttached(){
 	return false;
 }
 ReaderListener* CompoundReaderListener::getAttachedListener(){
+	std::lock_guard<std::mutex> lock(attached_listener_mutex);
```

Comment:
Yeah, at the time I was convinced it was necessary but I agree that it shouldn't be. I am having a hard time reproducing the crash today but I've removed it in https://github.com/eProsima/Fast-RTPS/pull/134/commits/2f3e68971a86f6c1fbfb50864d07a0d865799ac8 because if that was fixing it it was probably just a coincidence.